src/db/index.js

```diff
@@ -4,41 +4,56 @@ const Loki = require('lokijs');
 
 // our packages
 const {baseFolder} = require('../config');
-
-// TTL for auth requests
-const REQ_TTL =
-  process.env.NODE_ENV !== 'testing'
-    ? 5 * 60 * 1000 // 5 mins for prod
-    : 0; // 0 for tests
+const {reqCollection} = require('./requests');
+const logger = require('../logger');
 
 // init in-memory adapter for login requests
 const memAdapter = new Loki.LokiMemoryAdapter();
 const fsAdapter = new Loki.LokiFsAdapter();
 
-// init in-memory requests db
-const db = new Loki('requests.db', {adapter: memAdapter, autoload: true});
-// init persistent tokens db
+let dbLoadedResolve = () => {};
+const dbLoaded = new Promise(r => {
+  dbLoadedResolve = r;
+});
+
 let tokenCollection = {};
-const tokenDb = new Loki(path.join(baseFolder, 'auth.db'), {
+let secretsCollection = {};
+let deploysCollection = {};
+
+// init persistent db
+const db = new Loki(path.join(baseFolder, 'exoframe.db'), {
   adapter: process.env.NODE_ENV !== 'testing' ? fsAdapter : memAdapter,
   autoload: true,
   autoloadCallback() {
-    // get of create tokens collection
-    tokenCollection = tokenDb.getCollection('tokens');
+    // get or create tokens collection
+    tokenCollection = db.getCollection('tokens');
     if (tokenCollection === null) {
-      tokenCollection = tokenDb.addCollection('tokens');
+      tokenCollection = db.addCollection('tokens');
+    }
+
+    // get or create secrets collection
+    secretsCollection = db.getCollection('secrets');
+    if (secretsCollection === null) {
+      secretsCollection = db.addCollection('secrets');
+    }
+
+    // get or create deploys collection
+    deploysCollection = db.getCollection('deploys');
```

Comment:
I initially though of using file-based catalogue of deploys. 
I.e. you'd have a top folder `projectName` and nested 3-5 folders with dates as name corresponding to deployments.
This way you don't need to maintain a DB (and its relation to files) and can easily figure out when a deploy happened and what files need to be used to run it.
Thoughts?

Reply:
Okay, I will first try whether it works with files. Adding data to the database later doesn't sound too complicated to me.